python/tvm/contrib/debugger/debug_runtime.py

```diff
@@ -0,0 +1,424 @@
+            return
+
+        nodes_count = len(self.debug_datum.get_graph_nodes())
+        for i in range(nodes_count):
+            time_stamp = self._debug_run(i)
+            self.debug_datum.set_time(time_stamp)
+
+    def _run_debug(self):
+        """Invoke cli and when user execute select any operation,
+        'get_run_start_resp' return the user input.
+
+        Based on the user input, different type of Run will perform.
+        'set_debug_buffer' will set the empty buffer for setting node outputs.
+        Once the execution compled, output will be in the dump path and CLI will
+        be notified as run ends.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        None
+        """
+
+        #The ux may continue to execute multiple times, so after execution, give the control back
+        #to ux and it will decide when to stop
+        while True:
+            action = self.ui_obj.get_run_command()
+            if action == common.UxAction.DEBUG_RUN:
+                # Step 1. Set the debug buffer to catch the output
+                self._set_debug_buffer()
+                # Step 2. Execute the graph
+                retvals = self._debug_run_op_exec()
+                # Step 3. Dump the output tensors to the dump folder
+                self.debug_datum.dump_output_tensor(self.dbg_buff_list)
+                # Step 4. Inform ux execution completion.
+                action = self.ui_obj.run_end(action, retvals)
+            elif action == common.UxAction.NON_DEBUG_RUN:
+                retvals = super(GraphModuleDebug, self).run()
+                action = self.ui_obj.run_end(action, retvals)
+            else:
+                break
+            #If ux exits
+            if action == common.UxAction.EXIT:
+                break
+
+    def run(self, **input_dict):
+        """Run forward execution of the graph with debug
+
+        Parameters
+        ----------
+        input_dict : dict of str to NDArray
+            List of input values to be feed to
+        """
+        if input_dict:
+            self.set_input(**input_dict)
+        self._run_debug()
+
+    def set_input(self, key=None, value=None, **params):
+        """Set inputs to the module via kwargs
+
+        Along with the value setting to runtime, the same will be notified to
+        UI frontend as well.
+
+        Parameters
+        ----------
+
+        key : int or str
+           The input key
+
+        value : the input value.
+           The input key
+
+        params : dict of str to NDArray
+           Additonal arguments
+
+        Returns
+        -------
+        None
+        """
+        super(GraphModuleDebug, self).set_input(key, value, **params)
+
+        if key:
+            self.ui_obj.set_input(key, value)
+
+    def exit(self):
+        """Exits the dump folder and all its contents"""
+        self._remove_dump_root()
+
+class DebugGraphUXWrapper(object):
```

Comment:
Remove this class for now as it is unclear how this class can be used, alternatively, give an example on how to make simple UX out of the class